lib/compute/firewall.js

```diff
@@ -0,0 +1,259 @@
+    }
+    if (util.is(options.network, 'string')) {
+      this.network = options.network;
+    }
+    if (util.is(options.allowed, 'array')) {
+      this.allowed = options.allowed;
+    } 
+    if (util.is(options.sourceRanges, 'array')) {
+      this.sourceRanges = options.sourceRanges;
+    } 
+    if (util.is(options.sourceTags, 'array')) {
+      this.sourceTags = options.sourceTags;
+    } 
+    if (util.is(options.targetTags, 'array')) {
+      this.targetTags = options.targetTags;
+    } 
+  }
+}
+
+/**
+ * Delete the firewall rule.
+ *
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * firewall.delete(function(err) {});
+ */
+Firewall.prototype.delete = function(callback) {
+  callback = callback || util.noop;
+  this.makeReq_('DELETE', '', null, true, callback);
+};
+
+/**
+ * Update the firewall rule.
+ *
+ * @throws {Error} if firewall options are not provided, if no allowed 
+ *     protocols are specified or none of source ranges and source tags are 
+ *     provided
+ *
+ * @param {module:compute} compute - The Google Compute Engine instance this 
+ *     this firewall belongs to.
+ * @param {string} name - Name of the firewall.
+ * @param {object} options - Firewall options.
+ * @param {boolean} options.patch - If true update is performed according to
+ *     the patch semantics (default is false).
+ * @param {string=} options.description - Description of the firewall.
+ * @param {string=} options.network - Network to which the firewall applies. 
+ *     Default value is 'global/networks/default'.
+ * @param {object[]=} options.allowed - List of allowed protocols and ports.
+ * @param {string[]=} options.sourceRanges - IP address blocks to which this
+ *     rule applies (in CIDR format).
+ * @param {string[]=} options.sourceRanges - IP address blocks to which this
+ *     rule applies (in CIDR format).
+ * @param {string[]=} options.sourceTags - List of instance tags to which 
+ *     this rule applies.
+ * @param {string[]=} options.targetTags - List of instance tags indicating
+ *     instances that may process connections according to this rule.
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * var callback = function(err, firewall, apiResponse) {
+ *   // `firewall` is a Firewall object.
+ * };
+ *
+ * firewall.update(
+ *   {
+ *     patch: true,
+ *     description: 'Brand new description',
+ *   }, callback);
+ */
+Firewall.prototype.update = function(options, callback) {
+  if (!options) {
+    throw new Error('Firewall rule options must be provided.');
+  }
+
+  var method = 'PUT';
+  if (options.patch) {
+    method = 'PATCH';
+  } else {
+    if(!util.is(options.allowed, 'array')) {
+      throw new Error('Allowed protocols and ports must be provided.');
+    }
+    if (!util.is(options.sourceRanges, 'array') && 
+      !util.is(options.sourceTags, 'array')) {
+      throw new Error('Source ranges or source tags must be provided.');
+    }
+  }
+
+  var body = {};
```

Comment:
What is the use case for `PATCH`? I see where you're going with this since they essentially do the same thing, I'm just wondering if we actually need it or if it would better as its own method.